### PR TITLE
OHHHH SHITTTT ANTTTSSS

### PR DESCRIPTION
**Pull request by Fox-McCloud**
### **NOT AN APRIL FOOL'S PR**

Inspired by Goon's system where ants will spawn near food that has been laying out too long. Credit to them for the idea and the sprites!

Adds a system for ants infecting food!

If you leave food lying out and it's not on a table, eventually ants will appear on the food. Additionally, the food will be infected by ants. I really don't recommend eating food infested with ants...it could uh...be bad for you.

Overall, very simple and self-contained system; even with hundreds upon hundreds of food items the performance impact is non-existent.

How do you get rid of ants? Well, two ways: You can scoop them up or you can kill them with bugspray...or ya know, just not leave food lying all over the ground.

Refactor territory:
- Refactored it so all decals can have reagents, and not just cleanable decals
- Refactored decals to use a reagents list instead of snowflaked `reagents.add_reagent("farts", 10)`
- Renamed a few variables for readability

Pics:
img
img
Don't eat ants, you weirdo.

:cl: Fox McCloud
add: Adds space ants to the game; leave your food on tables or inside a container to avoid infestation!
/:cl:

### CODE REVIEW

File: code/game/objects/effects/effects.dm [hunk 1]
```diff
@@ -13,4 +13,38 @@
...
-	return FALSE
\ No newline at end of file
+	return FALSE
+
+/obj/effect/decal
+	var/no_scoop = FALSE   //if it has this, don't let it be scooped up
+	var/no_clear = FALSE    //if it has this, don't delete it when its' scooped up
+	var/list/scoop_reagents = null
+
+/obj/effect/decal/Initialize(mapload)
```
Comment (Fox-McCloud) [inline code/game/objects/effects/effects.dm]: We really need to convert all atoms to use `Initialize`, it's starting to cause problems.

Comment (Aurorablade) [reply code/game/objects/effects/effects.dm]: yeah we do but i can't remeber if its due to the fact we don't have all the subsystems in..orrrr just no ones done it...

**Comment by EvadableMoxie** [comment]
Do you want ants? Because this is how you get ants.

**Comment by TDSSS** [comment]
So we spray the food with bugspray to make it safe to eat again? Makes perfect sense.

**Comment by MrMagolor** [comment]
You should port the spiders chem, next. Of course, you'd have to change the recipe since it's a secret...

**Comment by davidchan13** [comment]
Does this apply to Botany plants that are harvested and left on the floor?

**Comment by Mitchs98** [comment]
If it applies to botany plants I can see it causing many problems and making botany worse than it is already due space issues, many people leave mutagen plants and such on the ground for ease of access. A table would solve it but...things are pretty cramped in there already.

**Comment by taukausanake** [comment]
This is neat and all but it just seems like a bit of fluff. There isn't much function to this, is there? Kind of boring to me but it's not bad

**Comment by Fox-McCloud** [comment]
>If it applies to botany plants I can see it causing many problems and making botany worse than it is already due space issues, many people leave mutagen plants and such on the ground for ease of access. A table would solve it but...things are pretty cramped in there already.

There's goodness knows how many plant bags lying around, not to mention the fridge for the chef; if you leave things lying around, that's on you.

If people are going to leave food lying around, then it's going to have consequences.

>Do ants crawl over food that would logically be wrapped up like chips and chocolate bars?

If you have any experience with ants, they tend to get into everything, even if it is wrapped up...especially if they're *space* ants.

**Comment by TDSSS** [comment]
>especially if they're space ants.

Bluespace ants, pls. They use the power of bluespace to get EVERYWHERE

**Comment by taukausanake** [comment]
If you're going to expand ants to anything besides what they are here make them something like the space worm or how spiderlings could have come if you tried mutating plants with bugs inside them. Or don't because self-antagging but then it's still kind of boring.

I was thinking of something like a mini-antag that's like a blob but it's just an ant hivemind instead. Trying to get to all the food it can. Kind of makes me with rats could do that and terrorize the chef

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 2]
```diff
@@ -11,6 +11,7 @@
 	possible_transfer_amounts = list(5,10,15,20,25,30,50)
 	volume = 50
 	burn_state = FIRE_PROOF
+	antable = FALSE
```
Comment (Lunaimon) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Var name seems a bit confusing.

Comment (KasparoVy) [reply code/modules/food_and_drinks/drinks/drinks.dm]: Is it antable, or an table? :thinking:

Comment (Fox-McCloud) [reply code/modules/food_and_drinks/drinks/drinks.dm]: `an table` is not correct grammar 😛

Comment (KasparoVy) [reply code/modules/food_and_drinks/drinks/drinks.dm]: Shiiiiit

**Comment by Fox-McCloud** [comment]
OHHHHHHHHHHHHHHHHHH SHITTTTTTTTTTTTTTT

🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜 🐜

**Pull request merged**